addons/stock_account/models/product.py

```diff
@@ -76,9 +76,9 @@ def _get_product_accounts(self):
         accounts = super(ProductTemplate, self)._get_product_accounts()
         res = self._get_asset_accounts()
         accounts.update({
-            'stock_input': res['stock_input'] or self.categ_id.property_stock_account_input_categ_id,
-            'stock_output': res['stock_output'] or self.categ_id.property_stock_account_output_categ_id,
-            'stock_valuation': self.categ_id.property_stock_valuation_account_id or False,
```

Comment:
Maybe use is_real_time instead of ` if self.categ_id.property_valuation == 'real_time'` each time ?


```suggestion
        accounts = super(ProductTemplate, self)._get_product_accounts()
        res = self._get_asset_accounts()
        is_real_time = self.categ_id.property_valuation == 'real_time'
        accounts.update({
            'stock_input': res['stock_input'] or self.categ_id.property_stock_account_input_categ_id if is_real_time else False,
            'stock_output': res['stock_output'] or self.categ_id.property_stock_account_output_categ_id if is_real_time else False,
            'stock_valuation': self.categ_id.property_stock_valuation_account_id if is_real_time else False,
```